.github/workflows/ci.yml

```diff
@@ -110,18 +110,42 @@ jobs:
     needs: [test]
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
     steps:
-      - uses: actions/checkout@v2
+      - name: Check out Git repository
+        uses: actions/checkout@v1
+
+      - name: Install Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 18
 
-      - uses: actions/download-artifact@v3
-        id: download
+      - name: Restore .cache
+        uses: actions/cache@v2
+        id: cache
         with:
-          name: built-on-ubuntu-latest
-          path: ~/downloads/
+          path: ${{ github.workspace }}/.cache
+          key: ${{ runner.os }}-${{ hashFiles('package*json', 'package-lock.json', '*config.js') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ hashFiles('package*json', 'package-lock.json', '*config.js') }}
+            ${{ runner.os }}-
+
+      - name: Restore node_modules
+        id: yarn-cache
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
```

Comment:
just FYI: aegir actually has a great github action that allows us to cache node-modules. we should be using that most places to ensure we're consistent.